### ISSUE TITLE
GVim [Wayland] window shift

### ISSUE THREAD

**Issue by phouchg42** [vim/vim#15517]
### Steps to reproduce

Start GVim with `env GVIM_ENABLE_WAYLAND=1`.

### Expected behaviour

If you start GVim with `env GVIM_ENABLE_WAYLAND=1` its window  should have equal indents on all sides, but it has strange shift as shown in the image (colored area, captured by GTK Inspector).

![Screenshot](https://github.com/user-attachments/assets/5432089c-7c4c-4875-bf14-23b72a35498f)

### Version of Vim

9.1.0672

### Environment

OS: Arch
DE: Gnome 46

**Comment by chrisbra** [comment]
I believe this happens, because the window is  slightly larger than what can be displayed using the number of rows/columns for your font size. I don't know how to make the window resize automatically to the required size.

**Comment by jmdevin** [comment]
> I don't know how to make the window resize automatically to the required size.

I don't believe it would be good to resize the window automatically even if it were possible; I have to think it would mess with users who start gvim maximized, or who use tiling window managers, etc.

**Comment by dvogel** [comment]
@jmdevin A few diagnostic questions:

* What is your `guioptions` value? 
* Does this occur if you start with `--clean`?
* Can you provide the output of `gvim --version`?

The gvim implementation actually does resize the window to match what is required by `Rows` and `Columns` unless `guioptions=k` is set.

For what it's worth, this is what I see with 9.1.0810 under GNOME 42 compiled against GTK3:

```
env GVIM_ENABLE_WAYLAND=1 ~/opt/vim/bin/vim -g --clean -background slateblue -c 'set guioptions='
```

![Screenshot from 2024-12-11 15-52-22](https://github.com/user-attachments/assets/fce997c1-ceaf-4c9c-a88d-fbb51fd8dcb6)

**Comment by moll** [comment]
Hey! I'm seeing the same thing on v9.1.1734 on Arch Linux and Wayland. I briefly alluded to _a_ width issue in https://github.com/vim/vim/issues/13575#issuecomment-3357524714, but the following invocation results in a window that's got a wider gap on the right side than on the left (as made visible via `colorcolumn`):

```
gvim --clean -c 'set guioptions= columns=81 colorcolumn=80 lines=25'
```

<img width="854" height="649" alt="Image" src="https://github.com/user-attachments/assets/813c7333-0972-496d-a0ec-4be94fbcbbc7" />

The gap persists if there's a right scrollbar. The aforementioned `GVIM_ENABLE_WAYLAND` doesn't seem to have any effect.

I'm also seeing Vim vertical size jump back to 40 lines when tabbing away from the newly opened window, but right now I suspect that's to do with GNOME's Wayland window manager not permitting window to change their size. Or perhaps that's also a Vim bug...

**Comment by ptr7F** [comment]
I’m experiencing a similar issue under Wayland. Certain font and font size configurations cause the bottom area to become unusually thick.

For example, in my screenshot, this happens when using the **Sarasa Fixed** font at size **11**. When I set the font size to **10** or **12**, the problem disappears.

This issue does not occur when running under X11.

**Wayland:**
<img width="1517" height="1294" alt="Image" src="https://github.com/user-attachments/assets/8d928358-5489-44e5-99d2-5c8a5868bd27" />


**x11:**
<img width="1517" height="1294" alt="Image" src="https://github.com/user-attachments/assets/0944a856-b749-4d89-b11e-f097e75e1fb2" />

**Comment by phouchg42** [comment]
Seems it’s not a GVim but something else (wayland, gtk4, gnome?) problem.
